### PR TITLE
config attribute supports dict

### PR DESCRIPTION
**Pull request by alexeagle**
Matt, you had the TODO to delete the `args` dict-attribute from esbuild, but it's used extensively in the test cases in rules_nodejs.
I think it's still nice to let the user pass a dict here, but we can syntax-sugar that into a standard config file easily in the macro. WDYT?

### CODE REVIEW

File: examples/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+esbuild(
+    name = "lib",
+    srcs = [
+        "main.js",
+        "name.js",
+    ],
+    entry_point = "main.js",
+    deps = [
+        # "@npm//date-fns",
```
Comment (mattem) [inline examples/BUILD.bazel:12]: Should this be included in deps? If not, remove the comment?

File: esbuild/repositories.bzl [hunk 2]
```diff
@@ -69,13 +69,12 @@ def esbuild_register_toolchains(name, esbuild_version, **kwargs):
...
         **kwargs: passed to each node_repositories call
     """
-    hashes = TOOL_VERSIONS[esbuild_version]
-    npm_import(
-        integrity = hashes["npm"],
-        package = "esbuild",
-        version = esbuild_version,
-    )
-
+    if esbuild_version not in TOOL_VERSIONS.keys():
```
Comment (mattem) [inline esbuild/repositories.bzl:72]: Seems unrelated to this PR?

File: esbuild/private/esbuild.bzl [hunk 3]
```diff
@@ -158,10 +153,10 @@ See https://esbuild.github.io/api/#target for more details
     """,
     ),
     "config": attr.label(
-        providers = [JSModuleInfo],
         mandatory = False,
-        doc = """Configuration file used for esbuild, from the esbuild_config macro. Note that options set in this file may get overwritten.
-See https://github.com/bazelbuild/rules_nodejs/tree/stable/packages/esbuild/test/plugins/BUILD.bazel for examples of using esbuild_config and plugins.
+        allow_single_file = True,
+        doc = """Configuration file used for esbuild. Note that options set in this file may get overwritten.
+        TODO: show how to write a config file that depends on plugins, similar to the esbuild_config macro in rules_nodejs.
```
Comment (mattem) [inline esbuild/private/esbuild.bzl:159]: Yeah I was going to comment that we should include that macro here
https://github.com/bazelbuild/rules_nodejs/blob/stable/packages/esbuild/test/plugins/BUILD.bazel#L21-L37

**Review by mattem — APPROVED** [review]
It would be nice if the unrelated parts are a different PR, but not going to block

File: esbuild/private/esbuild.bzl (hunk 3 above)
Comment (alexeagle) [reply esbuild/private/esbuild.bzl:159]: will file as an issue

**Review by alexeagle — COMMENTED** [review]
yes, I was lazy, it's now just the config attribute change and one motivating example
